src/main/java/seedu/contax/storage/CsvManager.java

```diff
@@ -24,28 +38,65 @@ public CsvManager(Model model, BiFunction<Integer, String[], Boolean> lineAction
     }
 
     /**
-     * Imports a given {@code IndexedCSvFile} to a {@code model}
+     * Imports a given {@code IndexedCsvFile} to a {@code model}
+     *
      * @param csvFile File to be imported
      * @return CommandResult upon success
      * @throws IOException if reading from file fails
      */
     // Return list of skipped lines
     public List<Integer> importCsv(IndexedCsvFile csvFile) throws IOException {
-        BufferedReader importedCsv = new BufferedReader(new FileReader(csvFile.getFilePath()));
+        BufferedReader importedCsvReader = new BufferedReader(new FileReader(csvFile.getFilePath()));
         String line = "";
         ArrayList<Integer> skippedLines = new ArrayList<>();
         int lineCounter = 0;
         //skip first line as it contains headers
-        importedCsv.readLine();
-        while ((line = importedCsv.readLine()) != null) {
+        importedCsvReader.readLine();
+        while ((line = importedCsvReader.readLine()) != null) {
             lineCounter++;
-            String[] importedPerson = line.split(",");
+            String[] importedPerson = line.split(",(?=([^\"]|\"[^\"]*\")*$)");
             boolean successful = lineAction.apply(lineCounter, importedPerson);
             if (!successful) {
                 skippedLines.add(lineCounter);
             }
         }
-        importedCsv.close();
+        importedCsvReader.close();
         return skippedLines;
     }
+
+    /**
+     * Exports the current address book into a pre-defined CSV file
+     *
+     * @return True if operation is successful, false otherwise
+     */
+    public boolean exportCsv(String filePath) {
+        try {
+            File exportedCsvFile = new File(filePath);
+            exportedCsvFile.createNewFile();
+            List<Person> listOfPersons = model.getFilteredPersonList();
+            BufferedWriter exportedCsvWriter = new BufferedWriter(new FileWriter(exportedCsvFile));
+            exportedCsvWriter.write(CSV_HEADER);
+            exportedCsvWriter.newLine();
+            for (Person person : listOfPersons) {
+                exportedCsvWriter.write(person.getName().toString() + ",");
+                exportedCsvWriter.write(person.getPhone().toString() + ",");
+                exportedCsvWriter.write(person.getEmail().toString() + ",");
+                if (person.getAddress().toString().contains(",")) {
+                    exportedCsvWriter.write("\"" + person.getAddress().toString() + "\",");
+                } else {
+                    exportedCsvWriter.write(person.getAddress().toString() + ",");
+                }
+                Set<Tag> personTags = person.getTags();
+                for (Tag tag : personTags) {
+                    exportedCsvWriter.write(tag.getTagName() + ";");
+                }
```

Comment:
Could you shift this part to the `ExportCsvCommand` logic instead? This is so that this file is pure with only I/O related functionality.

Reply:
thanks for the suggestion! will go ahead and implement this

Reply:
Updated, do take a look thanks!